include/ensmallen_bits/problems/dtlz/dtlz2_function.hpp

```diff
@@ -0,0 +1,199 @@
+/**
+ * @file dtlz2_function.hpp
+ * @author Satyam Shukla
+ *
+ * Implementation of the second DTLZ(Deb, Thiele, Laumanns, and Zitzler) test.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef ENSMALLEN_PROBLEMS_DTLZ_TWO_FUNCTION_HPP
+#define ENSMALLEN_PROBLEMS_DTLZ_TWO_FUNCTION_HPP
+
+#include "../../moead/weight_init_policies/uniform_init.hpp"
+
+namespace ens {
+namespace test {
+
+/**
+ * The DTLZ2 function, defined by:
+ * \f[
+ * x_M = [x_i, n - M + 1 <= i <= n]
+ * g(x) = \Sigma{i = n - M + 1}^n (x_i - 0.5)^2
+ * 
+ * f_1(x) = 0.5 * cos(x_1 * pi * 0.5) * cos(x_2 * pi * 0.5) * ... cos(x_2 * pi * 0.5) * (1 + g(x_M)) 
+ * f_2(x) = 0.5 * cos(x_1 * pi * 0.5) * cos(x_2 * pi * 0.5) * ... sin(x_M-1 * pi * 0.5) * (1 + g(x_M))
+ * .
+ * .
+ * f_M(x) = 0.5 * sin(x_1 * pi * 0.5) * (1 + g(x_M))
+ * \f]
+ *
+ * Bounds of the variable space is:
+ * 0 <= x_i <= 1 for i = 1,...,n.
+ *
+ * This should be optimized to x_i = 0.5 (for all x_i in x_M), at:
+ * 
+ * For more information, please refer to:
+ * 
+ * @code
+ * @incollection{deb2005scalable,
+ * title={Scalable test problems for evolutionary multiobjective optimization},
+ * author={Deb, Kalyanmoy and Thiele, Lothar and Laumanns, Marco and Zitzler, Eckart},
+ * booktitle={Evolutionary multiobjective optimization: theoretical advances and applications},
+ * pages={105--145},
+ * year={2005},
+ * publisher={Springer}
+ * }
+ * @endcode
+ *
+ * @tparam MatType Type of matrix to optimize.
+ */
+
+  template <typename MatType = arma::mat>
+  class DTLZ2
+  {
+    private:
+
+    // A fixed no. of Objectives and Variables(|x| = 7, M = 3).
+    size_t numObjectives {3};
+    size_t numVariables {7};
+    size_t numParetoPoints;
+
+    public:
+
+      DTLZ2 (size_t numParetoPoints = 136) :
```

Comment:
Can you add a function/parameter description? This applies to the other functions as well.